src/AbstractSQLRules2SQL.ts

```diff
@@ -939,6 +940,17 @@ const typeRules: Dictionary<MatchFn> = {
 		const date = DateValue(getAbstractSqlQuery(args, 0), indent);
 		return `DATE(${date})`;
 	},
+	DateTrunc: (args, indent) => {
+		if (engine !== Engines.postgres) {
+			throw new Error(
+				'DATE_TRUNC function is only supported on postgres currently',
+			);
+		}
+		checkArgs('DateTrunc', args, 2);
+		const str = TextValue(getAbstractSqlQuery(args, 0), indent);
+		const field = DateValue(getAbstractSqlQuery(args, 1), indent);
+		return `DATE_TRUNC(${str}, ${field})`;
```

Comment:
I'd recommend using names that match more closely to what they are so it's easier to know without looking at docs:
```suggestion
		const precision = TextValue(getAbstractSqlQuery(args, 0), indent);
		const date = DateValue(getAbstractSqlQuery(args, 1), indent);
		return `DATE_TRUNC(${precision}, ${date})`;
```